data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/producer/KafkaSinkProducer.java

```diff
@@ -54,57 +53,62 @@ public class KafkaSinkProducer<T> {
 
     private final DLQSink dlqSink;
 
-    private final CachedSchemaRegistryClient schemaRegistryClient;
-
     private final Collection<EventHandle> bufferedEventHandles;
 
     private final ExpressionEvaluator expressionEvaluator;
 
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
     private final String tagTargetKey;
 
-    private final ObjectMapper objectMapper = new ObjectMapper();
+    private final String topicName;
+
+    private final String serdeFormat;
+
+    private final SchemaService schemaService;
+
 
     public KafkaSinkProducer(final Producer producer,
                              final KafkaSinkConfig kafkaSinkConfig,
                              final DLQSink dlqSink,
-                             final CachedSchemaRegistryClient schemaRegistryClient,
                              final ExpressionEvaluator expressionEvaluator,
-                             final String tagTargetKey) {
+                             final String tagTargetKey
+    ) {
         this.producer = producer;
         this.kafkaSinkConfig = kafkaSinkConfig;
         this.dlqSink = dlqSink;
-        this.schemaRegistryClient = schemaRegistryClient;
         bufferedEventHandles = new LinkedList<>();
         this.expressionEvaluator = expressionEvaluator;
         this.tagTargetKey = tagTargetKey;
+        this.topicName = ObjectUtils.isEmpty(kafkaSinkConfig.getTopic()) ? null : kafkaSinkConfig.getTopic().getName();
+        this.serdeFormat = ObjectUtils.isEmpty(kafkaSinkConfig.getSerdeFormat()) ? null : kafkaSinkConfig.getSerdeFormat();
+        schemaService = new SchemaService.SchemaServiceBuilder().getFetchSchemaService(topicName, kafkaSinkConfig.getSchemaConfig()).build();
+
 
     }
 
     public void produceRecords(final Record<Event> record) {
         if (record.getData().getEventHandle() != null) {
             bufferedEventHandles.add(record.getData().getEventHandle());
         }
-        TopicConfig topic = kafkaSinkConfig.getTopic();
         Event event = getEvent(record);
         final String key = event.formatString(kafkaSinkConfig.getPartitionKey(), expressionEvaluator);
-        Object dataForDlq = event.toJsonString();
-        LOG.info("Producing record " + dataForDlq);
         try {
-            final String serdeFormat = kafkaSinkConfig.getSerdeFormat();
             if (MessageFormat.JSON.toString().equalsIgnoreCase(serdeFormat)) {
-                publishJsonMessage(record, topic, key, dataForDlq);
+                publishJsonMessage(record, key);
             } else if (MessageFormat.AVRO.toString().equalsIgnoreCase(serdeFormat)) {
-                publishAvroMessage(record, topic, key, dataForDlq);
+                publishAvroMessage(record, key);
```

Comment:
Why this check Why not `serdeFormat == MessageFormat.AVRO`?

Reply:
changes have been incorporated.